custom_components/truenas/__init__.py

```diff
@@ -227,6 +234,33 @@ def device_info(self):
         }
 
 
+class TrueNASPoolEntity:
+    """Represents a pool on the TrueNAS host."""
+
+    _pool: CachingPool
+
+    @property
+    def device_info(self):
+        return {
+            "identifiers": {
+                (DOMAIN, self._pool.guid),
+            },
+            "name": f"TrueNAS Pool - {self._pool.name}",
+        }
+
+    @property
+    def extra_state_attributes(self):
+        """Return extra Pool attributes"""
+        encryption = bool(self._pool.encrypt)
+        is_decrypted = bool(self._pool.is_decrypted)
+        return {
+            ATTR_POOL_NAME: f"{self._pool.name}",
+            ATTR_POOL_ID: f"{self._pool.id}",
+            ATTR_ENCRYPT: encryption,
+            ATTR_IS_DECRYPTED: is_decrypted,
```

Comment:
I'm fine having encryption as a state attribute here, but I think we should make `is_decrypted` a binary sensor in this integration instead (so you'd want to make a similar change that you made in `sensor.py` to `binary_sensor.py` to expose the encryption state).  We can [disable that new entity by default](https://developers.home-assistant.io/docs/entity_registry_disabled_by).